sklearn/isotonic.py

```diff
@@ -234,6 +234,32 @@ def __init__(self, y_min=None, y_max=None, increasing=True,
         self.increasing = increasing
         self.out_of_bounds = out_of_bounds
 
+    @property
+    @deprecated("Attribute ``X_`` is deprecated.")
```

Comment:
Please state the deprecation versions explicitly in the message:

"Attribute ``X_`` is deprecated in version 0.18 and will be removed in version 0.20."